File: Backend/DikkeTennisLijst.Core/Entities/Owned/Duration.cs
namespace DikkeTennisLijst.Core.Entities.Owned
{
    public class Duration
    {
        private Duration()
        {
        }

        public Duration(DateTimeOffset start, DateTimeOffset end)
        {
            if (start > end) throw new Exception("Start date is greater than end date.");
            Start = start;
            End = end;
        }

        public DateTimeOffset Start { get; set; }
        public DateTimeOffset End { get; set; }
        public int DurationInMinutes => (End - Start).Minutes;
    }
}

File: Frontend/frontend/src/app/_shared/entities/request/duration.ts
import { IDuration } from '@entities/response/iDuration';

export class Duration {
  constructor(start: Date, end: Date) {
    if (start > end) {
      throw new RangeError('Start date is greater than end date.');
    }
    this.start = start;
    this.end = end;

    this.durationInMinutes =
      this.end.getMinutes() +
      this.end.getHours() * 60 -
      this.start.getMinutes() -
      this.start.getHours() * 60;
  }

  public start: Date;
  public end: Date;
  public durationInMinutes: number;

  public static Create(duration: IDuration): Duration {
    return new Duration(new Date(duration.start), new Date(duration.end));
  }
}
